### DISCUSSION TITLE
Feature request: import Firefox bookmark JSON file

### DISCUSSION THREAD

**Discussion by shalak** [karakeep-app/karakeep#205] [Ideas]
It would be great to have a possibility of importing a JSON bookmark backup file from Firefox. It could even automatically add given bookmark to a list, based on the folder in which such bookmark is placed, as well as automatically import tags from the JSON export file.

**Comment by MohamedBassem** [comment]
While we currently doesn't support this from the Hoarder UI, have you considered using the CLI for doing such import? https://docs.hoarder.app/import/

**Reply by shalak** [reply]
Thanks for the tip! I only wish that we could interact with both lists and tags with the `bookmarks add` CLI, looks like it's not supported:

```
$ docker run --rm ghcr.io/hoarder-app/hoarder-cli:release bookmarks add --help
Usage: hoarder bookmarks add [options]

creates a new bookmark

Options:
  --link <link>  the link to add. Specify multiple times to add multiple links
                 (default: [])
  --note <note>  the note text to add. Specify multiple times to add multiple
                 notes (default: [])
  --stdin        reads the data from stdin and store it as a note
  -h, --help     display help for command

$ docker run --rm ghcr.io/hoarder-app/hoarder-cli:release bookmarks update --help
Usage: hoarder bookmarks update [options] <id>

update a bookmark

Arguments:
  id               the id of the bookmark to get

Options:
  --title <title>  if set, the bookmark's title will be updated
  --note <note>    if set, the bookmark's note will be updated
  --archive        if set, the bookmark will be archived
  --no-archive     if set, the bookmark will be unarchived
  --favourite      if set, the bookmark will be favourited
  --no-favourite   if set, the bookmark will be unfavourited
  -h, --help       display help for command
```

So I understand, that I would need to add a bookmark, obtain its ID, and then modify the list (no idea how to add a tag though, I don't see add option in `tags --help`).

Are there any plans for the `bookmark add` to take a list and tag as a parameter?

**Reply by MohamedBassem** [reply]
> that I would need to add a bookmark, obtain its ID, and then modify the list

Yeah, that's currently the way, not the most elegant way I agree.

> no idea how to add a tag though

Yeah, manipulating the tags seems missing from the CLI. Shouldn't be a hard addition.

> Are there any plans for the bookmark add to take a list and tag as a parameter

Sounds like a reasonable feature request :)

**Reply by kamtschatka** [reply]
I have created https://github.com/hoarder-app/hoarder/issues/209 to track this better
